src/tasks/publish/index.ts

```diff
@@ -0,0 +1,226 @@
+import { getMainContractName } from '../../utils/arappUtils'
+import * as apm from '~/src/utils/apm'
+import { generateArtifacts, validateArtifacts } from '~/src/utils/artifact'
+import { getFullAppName } from '~/src/utils/appName'
+import { ethers } from 'ethers'
+
+task(TASK_PUBLISH, 'Publish a new app version')
+  .addPositionalParam(
+    'bump',
+    'Type of bump (major, minor or patch) or semantic version',
+    undefined,
+    types.string
+  )
+  .addOptionalParam(
+    'contract',
+    'Contract address previously deployed.',
+    undefined,
+    types.string
+  )
+  .addOptionalParam(
+    'managerAddress',
+    'Owner of the APM repo. Must be provided in the initial release',
+    undefined,
+    types.string
+  )
+  .addOptionalParam(
+    'ipfsProvider',
+    'Provider URL to connect to an ipfs daemon API server',
+    'http://localhost:5001',
+    types.string
+  )
+  .addFlag(
+    'onlyContent',
+    'Prevents contract compilation, deployment and artifact generation.'
+  )
+  .addFlag('noVerify', 'Prevents etherscan verification.')
+  .setAction(async (params, bre: BuidlerRuntimeEnvironment) => {
+    // Do param type verification here and call publishTask with clean params
+    await publishTask(
+      {
+        bumpOrVersion: params.bump,
+        existingContractAddress: params.contract,
+        managerAddress: params.managerAddress,
+        ipfsProvider: params.ipfsProvider,
+        onlyContent: params.onlyContent,
+        noVerify: params.noVerify
+      },
+      bre
+    )
+  })
+
+async function publishTask(
+  {
+    bumpOrVersion,
+    existingContractAddress,
+    managerAddress,
+    ipfsProvider,
+    onlyContent,
+    noVerify
+  }: {
+    bumpOrVersion: string
+    existingContractAddress: string
+    managerAddress: string
+    ipfsProvider: string
+    onlyContent: boolean
+    noVerify: boolean
+  },
+  bre: BuidlerRuntimeEnvironment
+): Promise<void> {
+  const aragonConfig = bre.config.aragon as AragonConfig
+  const appSrcPath = aragonConfig.appSrcPath as string
+  const distPath = aragonConfig.appBuildOutputPath as string
+  const ipfsIgnore = ['subfolder/to/ignore/**'] // Hardcoded until a better way to deal with dynamic ENS address is found
+  const selectedNetwork = bre.network.name
+  // ### Todo: Add logic to guess app name from file?
+  const appName =
+    typeof aragonConfig.appName === 'string'
+      ? aragonConfig.appName
+      : typeof aragonConfig.appName === 'object'
+      ? aragonConfig.appName[selectedNetwork]
+      : undefined
+  if (!appName)
+    throw new BuidlerPluginError(`appName must be defined in buidler.config`)
+  const contractName = getMainContractName()
+
+  // Initialize clients
+  const networkConfig = bre.network.config as HttpNetworkConfig
+  const provider = new ethers.providers.Web3Provider(
+    bre.web3.currentProvider,
+    networkConfig.ensAddress && {
+      name: bre.network.name,
+      chainId: networkConfig.chainId || 5555,
```

Comment:
I might be missing context here—is `5555` the default chainId buidler or our ganache instance uses?

Reply:
That's definitely a magic number. I think that builder's type definitions require to pass a chainId so I have to pick one. Which number would you recommend?

Reply:
Gotcha; maybe it's something we could move to an environment or constants file with a comment?

We've had some weird issues in the past (2 years ago) with Metamask that forced us to randomize this value, but let's assume things have been fixed on their side until we see it pop back up.

Reply:
Add it in #77 as nice to have